extensions/avro-extensions/src/main/java/io/druid/data/input/AvroStreamInputRowParser.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.data.input;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.Pair;
+import io.druid.data.input.avro.GenericRecordAsMap;
+import io.druid.data.input.impl.ParseSpec;
+import io.druid.data.input.impl.TimestampSpec;
+import io.druid.data.input.schemarepo.SubjectAndIdConverter;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.util.ByteBufferInputStream;
+import org.joda.time.DateTime;
+import org.schemarepo.Repository;
+import org.schemarepo.api.TypedSchemaRepository;
+import org.schemarepo.api.converter.AvroSchemaConverter;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+
+public class AvroStreamInputRowParser<SUBJECT, ID> implements ByteBufferInputRowParser
+{
+  private final ParseSpec parseSpec;
+  private final List<String> dimensions;
+
+  private final TypedSchemaRepository<ID, Schema, SUBJECT> typedRepository;
+  private final SubjectAndIdConverter<SUBJECT, ID> subjectAndIdConverter;
+  private final Repository schemaRepository;
```

Comment:
I'm using the slightly different version of this row parser in production with schema repo(2 REST service behind nginx, 3-clusters-ZK backend) in production for about 10 months, no problem, our thoughput is about 300 kafka producers, 30 kafka consumers, 30 topics, rare schema revolution.

By saying "*this class is forcing every customer to use same.*" you mean druid customer?

I'm not guru with avro things, but after googling with "avro schema registry", I think there is not very much options, and this implemention is suitable for all(both) of search result in first page(schemarepo's and Confluent's)... Of course if you guys want support every possible deserialization from bytes to avro, I think abstracting it into `ByteBuffer -> GenericRecord` is the only way?